internal/pkg/agent/cmd/container.go

```diff
@@ -789,10 +789,9 @@ func setPaths(statePath, configPath, logsPath string, writePaths bool) error {
 	}
 	originalInstall := paths.Install()
 	originalTop := paths.Top()
-	paths.SetTop(topPath)
+	paths.SetTop(statePath) // change this?
```

Comment:
I think this is correct from my reading of the comments here: https://github.com/elastic/elastic-agent/blob/fc3eba3501a0695a477a46331293331873bb850c/internal/pkg/agent/application/paths/common.go#L59-L71

I also suspect you want to change the `topPath` variable here so that the permissions are set correctly:

https://github.com/elastic/elastic-agent/blob/fc3eba3501a0695a477a46331293331873bb850c/internal/pkg/agent/cmd/container.go#L769-L775